app/components/blog-form/ManagePostPage.js

```diff
@@ -9,41 +9,87 @@ class ManagePostPage extends React.Component {
     constructor(props, context) {
         super(props, context);
 
+        if (this.props.postId) { //If the request coming from the edit view
+            //Execute action to get post data to server
+            this.props.actions.getPost(this.props.postId);
+        }
+
         this.state = {
+            post: Object.assign({}, props.post),
             saving: false
         };
 
         //Mapping events
         this.handleSavePost = this.handleSavePost.bind(this);
     }
 
+    componentWillReceiveProps(nextProps){
+        if (this.props.post.id != nextProps.post.id){
+            //Required to populate form when existing course is loaded directly
+            this.setState({ post: Object.assign({}, nextProps.post) });
+        }
+
+        if (nextProps.post.error == 'Post not found'){ //Research if this is the best way to do this.
+            this.context.router.push('/app/blog');
+        }
+
+    }
+
     handleSavePost(post){
         this.setState({ saving: true });
-        this.props.actions.createPost(post)
-            .then(() => {
-                toastr.success('Post created successfully');
-                this.setState({ saving: false });
-                this.redirectOnSave();
-            })
-            .catch(error => {
-                toastr.error(error.description);
-                this.setState({ saving:false });
-            });
+
+        if (this.state.post.id == 0){ //New Post
+            this.props.actions.createPost(post)
+                .then(() => {
+                    toastr.success('Post created successfully');
+                    this.setState({ saving: false });
+                    this.redirectOnSave();
+                })
+                .catch(error => {
+                    toastr.error(error.description);
+                    this.setState({ saving:false });
+                });
+        } else { //Existing Post so -> Update Post
+            let postToUpdate = {
+                id: this.state.post.id,
+                title: post.title,
+                author: post.author,
+                content: post.content,
+                publishedDate: this.state.post.publishedDate
+            };
+            this.props.actions.updatePost(postToUpdate)
+                .then(() => {
+                    toastr.success('Post updated successfully');
+                    this.setState({ saving: false });
+                    this.redirectOnSave();
+                })
+                .catch(error => {
+                    toastr.error(error.description);
+                    this.setState({ saving:false });
+                });
+        }
+
```

Comment:
I think we should have a separated form for update, using the same form adds more logic, and it gets complicated to maintain and declouple